src/blame.rs

```diff
@@ -0,0 +1,512 @@
+  /// @returns A new Blame object for the buffer content
+  /// @throws If the buffer contains invalid UTF-8
+  pub fn buffer(&self, buffer: Buffer, buffer_len: u32, env: Env) -> Result<Blame> {
+    let content = std::str::from_utf8(&buffer[..buffer_len as usize])
+      .map_err(|e| Error::from_reason(format!("Invalid UTF-8 in buffer: {}", e)))?;
+
+    let blame = match &self.inner {
+      BlameInner::Repo(shared_ref) => {
+        let cloned = shared_ref.clone(env)?;
+
+        cloned.share_with(env, |git_blame| {
+          git_blame
+            .blame_buffer(content.as_bytes())
+            .map_err(|e| Error::from(crate::Error::from(e)))
+        })?
+      }
+    };
+
+    Ok(Blame {
+      inner: BlameInner::Repo(blame),
+    })
+  }
+
+  #[napi]
+  /// Gets blame information for the specified index
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   getHunkByIndex(index: number): BlameHunk;
+  /// }
+  /// ```
+  ///
+  /// @param {number} index - Index of the hunk to get (0-based)
+  /// @returns Blame information for the specified index
+  /// @throws If no hunk is found at the index
+  pub fn get_hunk_by_index(&self, index: u32) -> Result<BlameHunk> {
+    let hunk = self
+      .inner
+      .get_index(index as usize)
+      .ok_or_else(|| Error::from_reason(format!("No blame hunk found at index {}", index)))?;
+
+    let commit_id = hunk.final_commit_id().to_string();
+    let is_zero_commit = commit_id == "0000000000000000000000000000000000000000";
+
+    let signature = if is_zero_commit {
+      None
+    } else {
+      Signature::try_from(hunk.final_signature()).ok()
+    };
+
+    let path = hunk.path().map(|p| p.to_string_lossy().to_string());
+
+    Ok(BlameHunk {
+      commit_id,
+      final_start_line_number: hunk.final_start_line() as u32,
+      lines_in_hunk: hunk.lines_in_hunk() as u32,
+      signature,
+      path,
+      orig_start_line_number: hunk.orig_start_line() as u32,
+      is_boundary: hunk.is_boundary(),
+    })
+  }
+
+  #[napi]
+  /// Gets blame information for the specified line
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   getHunkByLine(line: number): BlameHunk;
+  /// }
+  /// ```
+  ///
+  /// @param {number} line - Line number to get blame information for (1-based)
+  /// @returns Blame information for the specified line
+  /// @throws If no hunk is found for the line
+  pub fn get_hunk_by_line(&self, line: u32) -> Result<BlameHunk> {
+    let hunk = self
+      .inner
+      .get_line(line as usize)
+      .ok_or_else(|| Error::from_reason(format!("No blame hunk found for line {}", line)))?;
+
+    let commit_id = hunk.final_commit_id().to_string();
+    let is_zero_commit = commit_id == "0000000000000000000000000000000000000000";
+
+    let signature = if is_zero_commit {
+      None
+    } else {
+      Signature::try_from(hunk.final_signature()).ok()
+    };
```

Comment:
```suggestion
    let commit_id = hunk.final_commit_id();
    let signature = if commit_id.is_zero() {
      None
    } else {
      Signature::try_from(hunk.final_signature()).ok()
    };
```
How about using the `is_zero` method implemented in Oid?